hca/util/__init__.py

```diff
@@ -235,29 +238,29 @@ def store_refs(d):
 
     @property
     def swagger_spec(self):
-        if not self._swagger_spec:
-            swagger_url = self.config[self.__class__.__name__].swagger_url
-            if "swagger_filename" in self.config:
-                swagger_filename = self.config.swagger_filename
-                if not swagger_filename.startswith("/"):
-                    swagger_filename = os.path.join(os.path.dirname(__file__), swagger_filename)
-            else:
-                swagger_filename = base64.urlsafe_b64encode(swagger_url.encode()).decode() + ".json"
-                swagger_filename = os.path.join(self.config.user_config_dir, swagger_filename)
-            if (("swagger_filename" not in self.config) and
-                ((not os.path.exists(swagger_filename)) or
-                 (fs.get_days_since_last_modified(swagger_filename) >= self._spec_valid_for_days))):
-                try:
-                    os.makedirs(self.config.user_config_dir)
-                except OSError as e:
-                    if not (e.errno == errno.EEXIST and os.path.isdir(self.config.user_config_dir)):
-                        raise
-                res = self.get_session().get(swagger_url)
-                res.raise_for_status()
-                assert "swagger" in res.json()
-                fs.atomic_write(os.path.basename(swagger_filename), self.config.user_config_dir, res.content)
-            with open(swagger_filename) as fh:
-                self.__class__._swagger_spec = self.load_swagger_json(fh)
+        with self.__class__._swagger_spec_lock:
```

Comment:
I'm not a fan of `self.__class__.attribute`. It's mildly paranoid. Class attributes are accessible directly on self.